src/app/main.rs

```diff
@@ -196,6 +195,22 @@ where
             });
 
         let (tap_layer, tap_grpc, tap_daemon) = tap::new();
+        let (tap_http_metrics, tap_http_report) =
+            http_metrics::new::<TapLabels, Class>(config.metrics_retain_idle);
+
+
+        let tap_http_server = shared::stack(::api::tap::server::TapServer::new(tap_grpc))
+            .push(phantom_data::layer())
+            .push(proxy::grpc::server::layer())
+            .push(http_metrics::layer::<_, classify::Response>(
+                tap_http_metrics,
+            ))
```

Comment:
Actually, without a `classify::layer()`, would this ever report metrics?

Reply:
Or... does the `classify::layer` provide the labels, and we've already done so with `TapLabels`?

Reply:
https://github.com/linkerd/linkerd2-proxy/blob/2f2050537dd8caeaf5704e66a3fd7be89f2455a3/src/proxy/http/metrics/service.rs#L250

A classify layer is only necessary when it needs configuration. When using the default logic, the metrics layer uses a default.